### PR TITLE
ParameterizedJobMixIn  for workflow

### PR DESCRIPTION
**Pull request by KostyaSha**
@jglick 
Or how do you expect running workflow builds with actions?

### CODE REVIEW

**Comment by jglick** [comment]
What are you trying to solve here? Code sample, matching downstream PR? The `scheduleBuild2` helper method already allows jobs to be scheduled with actions.

**Comment by KostyaSha** [comment]
@jglick how get quiet time for scheduling build with Actions then? asJob is not public

**Comment by KostyaSha** [comment]
> Code sample, matching downstream PR?

https://github.com/jenkinsci/github-pullrequest-plugin/pull/26/files#diff-7068f23747266083876e3cd6073e4aa2R143

**Comment by jglick** [comment]
You can just pass `-1`.

**Comment by KostyaSha** [comment]
-1 and new method will work.

As i see `getBuildNowText();` is exposed, but `getQuietPeriod()` doesn't.

**Comment by oleg-nenashev** [comment]
The pull request still needs more justification. It includes new API method (getQuitePeriod()) + some extra payload. I would rather fix it.

**Comment by KostyaSha** [comment]
@oleg-nenashev Sorry, fix what? Suitable method was added in API few releases ago, -1 LTS can't allow running workflow jobs workflow that claimed to be nearest feature. 
Of course it can be my pure java knowledge (this code looks like spaghetti) but i see no way to get `quietPeriod`. Multi-argument method (ala win api) resolves part of problem, but i see no comments to remove new API method that i introduced in PR. Why other methods was added then? It looks like jenkins repeating mistakes by adding pure APIs/interfaces. 

Protected methods/fields already created many copy-pastes and hacks in plugins. Is it make sense to make `asJob()` public?

**Comment by KostyaSha** [comment]
Blocker for workflow integration.

File: core/src/main/java/jenkins/model/ParameterizedJobMixIn.java [hunk 1]
```diff
@@ -138,7 +145,7 @@ public final boolean scheduleBuild(int quietPeriod, Cause c) {
         return Jenkins.getInstance().getQueue().schedule2(asJob(), quietPeriod, queueActions).getItem();
     }
 
-    private List<ParameterValue> getDefaultParametersValues() {
+    public List<ParameterValue> getDefaultParametersValues() {
```
Comment (jglick) [inline core/src/main/java/jenkins/model/ParameterizedJobMixIn.java]: Does this really need to be public? Why?

**Comment by jglick** [comment]
Still do not follow the motivation. Other plugins have managed to schedule builds with no loss of functionality AFAIK. PR description does not justify the changes.

File: core/src/main/java/jenkins/model/ParameterizedJobMixIn.java (hunk 1 above)
Comment (KostyaSha) [inline core/src/main/java/jenkins/model/ParameterizedJobMixIn.java]: All plugins copy-pasting it and no reason to have it private
https://github.com/jenkinsci/github-pullrequest-plugin/blob/6e6e457352d094b916b090972464ad401f1b2a46/src/main/java/org/jenkinsci/plugins/github/pullrequest/trigger/JobRunnerForCause.java#L123-L150

https://github.com/jenkinsci/dockerhub-notification-plugin/blob/6bd035ecfa2b8e46b7ef475c088668636e89b017/src/main/java/org/jenkinsci/plugins/dockerhub/notification/DockerHubWebHook.java#L198

https://github.com/jenkinsci/gerrit-trigger-plugin/blob/2224079b54947ddf8e90284e3ec9a8d0866d5b78/src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/EventListener.java#L259

https://github.com/search?q=user%3Ajenkinsci++%22getDefaultParametersValues%22&ref=searchresults&type=Code&utf8=%E2%9C%93

**Comment by KostyaSha** [comment]
@jglick probably they schedule with `0` but `quietPeriod` belongs to standard job property of `parametrizedJobMixIn` so:
1) API should somehow provide ability to get it. 
2) I can remove new method as multi arg with `-1` allows run what i want. Though it not clear then why other single arg methods was introduced.

I don't think that it easy for plugin implementations making custom classes with overloading this protected methods 
https://github.com/jenkinsci/dockerhub-notification-plugin/blob/6bd035ecfa2b8e46b7ef475c088668636e89b017/src/main/java/org/jenkinsci/plugins/dockerhub/notification/DockerHubWebHook.java#L205-L231
Seems i can copy-paste this work-around until core API expose methods.

**Comment by jglick** [comment]
I have no idea what the code in `DockerHubWebHook` is doing, perhaps @jtnord can explain it.

**Comment by KostyaSha** [comment]
@jglick all plugins trying expose required methods that are protected

**Comment by jtnord** [comment]
@jglick `DockerHubWebHook` is nothing to do with me (I just cleaned the code up and removed the proprietary information for open sourcing).  @rsandell should know more.

File: core/src/main/java/jenkins/model/ParameterizedJobMixIn.java (hunk 1 above)
Comment (oleg-nenashev) [inline core/src/main/java/jenkins/model/ParameterizedJobMixIn.java]: The change is reasonable IMO. Needs Javadoc BTW

**Comment by oleg-nenashev** [comment]
Weak :+1: from me

File: core/src/main/java/jenkins/model/ParameterizedJobMixIn.java (hunk 1 above)
Comment (rsandell) [inline core/src/main/java/jenkins/model/ParameterizedJobMixIn.java]: Yea, I should have made that public way back when I needed it, but I wasn't a core committer then and after I just forgot, and when I implemented the dockerhub trigger I remembered that I needed it, and forgot it again.

Basically the plugins wants to add their own parameters to the build, but not override the already defined parameters and add their default values to the build as well when triggering.

**Comment by KostyaSha** [comment]
Sorry what is the summary? What should i do?

**Comment by oleg-nenashev** [comment]
At least Javadocs for the public method

**Comment by KostyaSha** [comment]
Thanks for reviews, don't work anymore on this.

**Pull request closed without merging**